Latest Superpowered

JazariMusic opened the pull request.

ivannador committed:
Fixed erroneous retain by NSTimer

ivannador committed:
Merge pull request #794 from superpoweredSDK/ivannador/ios-audioio-retain-fix

Fixed erroneous retain by NSTimer

ivannador committed:
Updated React Native example to 0.78, added Turbo native module

ivannador committed:
Merge pull request #796 from superpoweredSDK/ivannador/react-native-example-update

Updated React Native example to 0.78, added Turbo native module

ivannador committed:
Merge pull request #798 from superpoweredSDK/release/2.7.3

Version 2.7.3

ivannador committed:
Version 2.7.4 (#800)

ivannador committed:
Version 2.7.5 (#804)

* Version bump

* Windows builds

* Apple builds

* Modify VisionOS example

* Android builds

* Windows rebuild

* Linux builds

Commit:
Merge remote-tracking branch 'upstream/master' into james/merge-new-superpowered

JazariMusic commented on Superpowered/OpenSource/SuperpoweredIOSAudioIO.mm:
```diff
@@ -113,9 +113,31 @@ - (id)initWithDelegate:(NSObject<SuperpoweredIOSAudioIODelegate> *)d
 #if (USES_AUDIO_INPUT == 1)
```
This is an important file to review @tehsven @cwalo

tehsven replied:
lol
<img width="5120" height="1860" alt="Screenshot 2025-07-15 at 10 36 28" src="https://github.com/user-attachments/assets/3510c9f8-fd35-468d-9b55-cc0aab552b2b" />

tehsven replied:
@JazariMusic pushed a fix for this: https://github.com/ResonantCavity/Low-Latency-Android-Audio-iOS-Audio-Engine/compare/22b5ab2aa695...d8ca748dac62

JazariMusic commented on Superpowered/libSuperpoweredAudio.xcframework/Info.plist:
```diff
@@ -6,79 +6,83 @@
+			<string>macos-arm64_x86_64</string>
 			<key>LibraryPath</key>
-			<string>libSuperpoweredAudioIOS-iphoneos.a</string>
+			<string>libSuperpoweredAudio.a</string>
```
This name change may have build consequences.

tehsven replied:
Good catch. This is fine though since the iOS project references the framework, not specific binaries.

<img width="1832" height="1756" alt="Screenshot 2025-07-15 at 11 10 42" src="https://github.com/user-attachments/assets/11fba9c9-6420-48b1-adbd-7310b7295d40" />

pmarino reviewed (approved):
👍 Ran some smoke tests and tested all presets. Everything appears to be working as expected. Also, the 2.7.5 update does indeed fix the decoder deadlock issue.

tehsven committed:
Use direct class type instead of typeof to avoid compilation failure

typeof isn't an objective-c keyword, and unless you use specific compiler
settings, using typeof won't build.

Using the specific type of the class works without trying to add typeof
support to the build pipeline.

tehsven reviewed (approved):
After pushing the `typeof` fix, I was able to smoke test some multitrack playback/recording, and was able to make an archive.
